lib/shared/gen.js

```diff
@@ -116,8 +117,16 @@ module.exports = {
         return Promise.resolve('export default {' + Object.entries(setup).map(([key, origValue]) => {
             let value = JSON.stringify(origValue);
 
-            if (key === 'data' && typeof origValue !== 'string') {
-                value = `JSON.parse(${JSON.stringify(value)})`;
+            if (key === 'mode') {
+                value = `function ${getModePlaceholder()}(){return${value}}()`;
```

Comment:
It's better to add a space after `return`. In case value is a number, `null`, `false` or `true` it will break.

Reply:
Fixed